Shoko.Server/API/v3/Models/Shoko/Renamer.cs

```diff
@@ -0,0 +1,119 @@
+            Enabled = ServerSettings.Instance.Plugins.EnabledRenamers.ContainsKey(Name) ? ServerSettings.Instance.Plugins.EnabledRenamers[Name] : true;
+        }
+
+        /// <summary>
+        /// Renamer ID.
+        /// </summary>
+        public int ID { get; set; }
+
+
+        /// <summary>
+        /// A short description about the renamer.
+        /// </summary>
+        public string Description { get; set; }
+
+
+        /// <summary>
+        /// If the renamer is enabled.
+        /// </summary>
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Lower numbers mean higher priority. Will be null if a priority is not set yet.
+        /// </summary>
+        public int? Priority { get; set; }
+
+        public class Script
+        {
+            public Script()
+            {
+            }
+
+            public Script(RenameScript script)
+            {
+                RenameFileHelper.RenamerInfo renamerInfo;
+                RenameFileHelper.Renamers.TryGetValue(script.RenamerType, out renamerInfo);
+                IDs = new RenamerScriptIDs() { ID = script.RenameScriptID, Renamer = renamerInfo?.ID };
+                RenamerName  =  script.RenamerType;
+                Name = script.ScriptName;
+                EnabledOnImport = script.IsEnabledOnImport == 1;
+                Body = script.Script;
+            }
+
+            public void MergeWithExisting(RenameScript existing)
+            {
+                if (IDs.Renamer.HasValue)
+                {
+                    var renamerInfo = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == IDs.Renamer.Value);
+                    if (renamerInfo != null)
+                        existing.RenamerType = renamerInfo.Name;
+                }
+                existing.Script = Body;
+                existing.ScriptName = Name;
+                existing.IsEnabledOnImport = EnabledOnImport ? 1 : 0;
+            }
+
+            /// <summary>
+            /// Script IDs.
+            /// </summary>
+            [Required]
+            public RenamerScriptIDs IDs { get; set; }
+
+            /// <summary>
+            /// The last known name of the <see cref="Models.Shoko.Renamer"/> this script belongs to.
+            /// </summary>
+            public string RenamerName { get; set; }
+
+            /// <summary>
+            /// Script name. Must be unique across all scripts.
+            /// </summary>
+            [Required]
+            public string Name { get; set; }
+
+            /// <summary>
+            /// Determines if the script should ran automatically on import if the renamer is enabled.
+            /// </summary>
+            [Required]
+            public bool EnabledOnImport { get; set; }
+
+            /// <summary>
+            /// The script body.
+            /// </summary>
+            [Required(AllowEmptyStrings = true)]
+            public string Body { get; set; }
```

Comment:
This should be nullable, and not required.

Reply:
Fixed.